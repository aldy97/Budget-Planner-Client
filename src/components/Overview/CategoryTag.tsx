import React from "react";
import { COLORS } from "../../utils/constants";
import { Tag } from "antd";

interface TagProps {
  category: string;
  type: string;
}

function CategoryTag({ category, type }: TagProps): JSX.Element {
  return (
    <Tag color={type === "expense" ? COLORS.ORANGE : COLORS.THEMEBLUE}>
      {category}
    </Tag>
  );
}

export default CategoryTag;
