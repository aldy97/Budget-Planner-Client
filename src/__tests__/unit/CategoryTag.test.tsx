import React from "react";
import { mount } from "enzyme";
import CategoryTag from "../../components/Overview/CategoryTag";

describe("CategoryTag", () => {
  const wrapper = mount(<CategoryTag type="expense" category="test"></CategoryTag>);

  it("can render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("can receive its props", () => {
    expect(wrapper.prop("type")).toEqual("expense");
    expect(wrapper.prop("category")).toEqual("test");
  });
});
