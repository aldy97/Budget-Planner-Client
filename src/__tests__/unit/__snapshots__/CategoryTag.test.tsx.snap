// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTag can render without crashing 1`] = `ReactWrapper {}`;
